test-suite/Djinni.cmake

```diff
@@ -1,13 +1,14 @@
-
-# Resolve DJINNI_EXECUTABLE
-set(DJINNI_EXECUTABLE_NAMES "djinni")
-find_program(DJINNI_EXECUTABLE ${DJINNI_EXECUTABLE_NAMES}
-  DOC "Path to the Djinni executable"
-  PATHS ${DJINNI_EXECUTABLE_PATH})
-if(DJINNI_EXECUTABLE)
-  message(STATUS "Djinni executable: ${DJINNI_EXECUTABLE}")
+cmake_minimum_required(VERSION 3.18)
+
+# find Djinni executable.
+# On windows find_program() does not work for finding the `djinni.bat` script.
+# The script must either be on the PATH or `DJINNI_EXECUTABLE` must explicitly be predefined.
+if(CMAKE_HOST_SYSTEM_NAME STREQUAL "Windows")
+  if(NOT DEFINED CACHE{DJINNI_EXECUTABLE})
+    set(DJINNI_EXECUTABLE djinni.bat CACHE FILEPATH "path of djinni binary")
+  endif()
 else()
-  message(FATAL_ERROR "Could not find DJINNI_EXECUTABLE using the following names: ${DJINNI_EXECUTABLE_NAMES}")
+  find_program(DJINNI_EXECUTABLE djinni REQUIRED)
```

Comment:
`find_program(... REQUIRED)` automatically throws an error if the program cannot be found. That way the custom check (`if(DJINNI_EXECUTABLE`) is not required any more. Requires cmake >= 3.18

Reply:
I haven't explicitly tested it but my understandig is the same. Only difference to the "old" function should be that it automatically produces an error if the program cannot be found.